game-app/game-core/src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -1281,21 +1287,24 @@ public static Predicate<Territory> isTerritoryEnemyAndNotUnownedWater(
   }
 
   public static Predicate<Territory> isTerritoryEnemyAndNotUnownedWaterOrImpassableOrRestricted(
-      final GamePlayer player,
-      final GameProperties properties,
-      final RelationshipTracker relationshipTracker) {
-    return t -> {
-      if (t.isOwnedBy(player)) {
-        return false;
-      }
-      // if we look at territory attachments, may have funny results for blockades or other things
-      // that are passable and not owned. better to check them by alliance. (veqryn)
-      if (t.isOwnedBy(GamePlayer.NULL_PLAYERID) && t.isWater()) {
-        return false;
-      }
-      return territoryIsPassableAndNotRestricted(player, properties).test(t)
-          && relationshipTracker.isAtWar(player, t.getOwner());
-    };
+      final GamePlayer player) {
+    final RelationshipTracker relationshipTracker = player.getData().getRelationshipTracker();
+    return territoryNotImpassibleOrRestrictedOrNeutralWaterAndNotOwnedBy(player)
+        .and(t -> relationshipTracker.isAtWar(player, t.getOwner()));
+  }
+
+  public static Predicate<Territory> isTerritoryNotUnownedWaterAndCanBeTakenOverBy(
+      final GamePlayer player) {
+    final RelationshipTracker relationshipTracker = player.getData().getRelationshipTracker();
+    return territoryNotImpassibleOrRestrictedOrNeutralWaterAndNotOwnedBy(player)
+        .and(t -> relationshipTracker.canTakeOverOwnedTerritory(player, t.getOwner()));
+  }
+
+  private static Predicate<Territory> territoryNotImpassibleOrRestrictedOrNeutralWaterAndNotOwnedBy(
+      final GamePlayer player) {
+    return not(isTerritoryOwnedBy(player))
+        .and(not(territoryIsUnownedWater()))
+        .and(territoryIsPassableAndNotRestricted(player, player.getData().getProperties()));
```

Comment:
If you're simplifying the method signature anyways, would it be a benefit to have `territoryIsPassableAndNotRestricted` only accept a `GamePlayer` and derive the properties from that? Or is this just a special case that happens only here?

Reply:
One danger is if the null player instance is ever passed, that one will not have a null data. I filed an issue about getting rid of a global null player instance and instead having a null instance associated with the game data, but that's a bigger refactoring.

So it's not necessarily safe to do - depends on call sites.

Reply:
Keeping as-is per the above. Once we can change the engine such that the null player is guaranteed to have a non-null data, I'd be more comfortable refactoring broader in this way.